### PR TITLE
Clean BuildFiles in reco plugins

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick partially automatic BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/30139), this time covering all `use`s in the plugins which are not directly in the root of the BuildFile. My script did not catch these before.

As always, only the dependencies which are **not included in any of the plugin sources** are removed, so these changes are orthogonal and complementary to the recent PRs which added all packages that are actually included in the **libraries**.

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by pohsun** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

File: RecoHI/HiTracking/test/BuildFile.xml [hunk 1]
```diff
@@ -4,7 +4,6 @@
 <use name="Geometry/CommonDetUnit"/>
 <use name="Geometry/Records"/>
 <use name="Geometry/TrackerGeometryBuilder"/>
-<use name="DataFormats/TrackReco"/>
```
Comment (slava77) [inline RecoHI/HiTracking/test/BuildFile.xml:7]: this was added recently in #30020  .
Looking at RecoHI/HiTracking/test/HIPixelClusterVtxAnalyzer.cc
I suspect that there in #30020 DataFormats/TrackReco somehow came though an include of DataFormats/VertexReco, which is actually not used.
To be safe(er), please edit RecoHI/HiTracking/test/HIPixelClusterVtxAnalyzer.cc to get rid of DataFormats/VertexReco includes.

File: RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-
+#include "DataFormats/L1GlobalTrigger/interface/L1GlobalTriggerEvmReadoutRecord.h"
```
Comment (slava77) [inline RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:1]: why is this needed here?
This include is present in RecoVertex/BeamSpotProducer/interface/BeamSpotOnlineProducer.h

Comment (guitargeek) [reply RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:1]: Because otherwise my checker would say that `DataFormats/L1GlobalTrigger` is not needed, because it considers `interface` to not be a part of the plugins. But the best solution is probably to merge the header file with the source, I'll do that.

File: RecoHI/HiTracking/test/BuildFile.xml (hunk 1 above)
Comment (guitargeek) [reply RecoHI/HiTracking/test/BuildFile.xml:7]: Good catch, thanks!

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by jfernan2** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/30535	02c5831
- code changes are technical, they are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no differences

**Comment by kpedro88** [comment]
+upgrade

**Comment by Martin-Grunewald** [comment]
There is now a conflict with BeamSpotOnlineProducer.h and .cc - why are those files modified in this PR anyway?

**Comment by guitargeek** [comment]
Thanks for the reminder! These files are modified because if the `interface` directory contains both the headers for the plugins and the library, disentangling the dependencies is not possible (see https://github.com/cms-sw/cmssw/pull/30535#discussion_r450549406). Hence the `.h` and `.cc` files had to be merged to avoid false positives in the flagging of unused dependencies.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by jfernan2** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by pohsun** [comment]
+1

File: RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc [hunk 3]
```diff
@@ -1,10 +1,47 @@
...
+#include "DataFormats/Scalers/interface/BeamSpotOnline.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+#include "CondFormats/DataRecord/interface/BeamSpotObjectsRcd.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
 #include "FWCore/Framework/interface/EventSetup.h"
-#include "FWCore/Framework/interface/IOVSyncValue.h"
-#include "FWCore/Framework/interface/MakerMacros.h"
+//#include "FWCore/Framework/interface/IOVSyncValue.h"
```
Comment (slava77) [inline RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:21]: is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Comment (slava77) [reply RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc:21]: ... this is new compared to the previously signed version by reco

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/30535	dbff213
- considering 4 signatures out of 6 already collected, https://github.com/cms-sw/cmssw/pull/30535#discussion_r453792044 can be resolved in a follow-up PR

**Comment by Martin-Grunewald** [comment]
+1

**Comment by silviodonato** [comment]
+1
